### PR TITLE
RFC 311: Add site-admin UI page to view migrations

### PR DESCRIPTION
**Pull request by efritz**
This adds a UI Page to site-admin that shows the current progress of out-of-band migrations. This fixes https://github.com/sourcegraph/sourcegraph/issues/18239 and I'm pushing this to get out to make it obvious when out of band migrations are not working properly (and why!) in our cloud environments as we're getting an increasing number of users.

Current development environment screenshots:
Screen Shot 2021-03-31 at 4 56 25 PM

Storyboard screenshots:
Screen Shot 2021-03-31 at 4 54 49 PM
Screen Shot 2021-03-31 at 4 54 54 PM
Screen Shot 2021-03-31 at 4 55 38 PM
Screen Shot 2021-03-31 at 4 55 44 PM
Screen Shot 2021-03-31 at 4 55 49 PM

I'd like to rope in @sourcegraph/design once they have the bandwidth to work on more internal-facing (developer troubleshooting) areas of the application.

### CODE REVIEW

File: client/web/src/site-admin/SiteAdminMigrationsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+    const match = version.match(/^(\d+)\.(\d+)(\.|$)/)
+    if (!match) {
+        return undefined
+    }
+
+    return {
+        major: parseInt(match[1], 10),
+        minor: parseInt(match[2], 10),
+    }
+}
```
Comment (eseliger) [inline client/web/src/site-admin/SiteAdminMigrationsPage.tsx:400]: I think we have the semver package pulled in somewhere else anyways, so you could also just use that

File: client/web/src/site-admin/SiteAdminMigrationsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+        useMemo(
+            () =>
+                timer(0, REFRESH_INTERVAL_MS, undefined).pipe(
+                    concatMap(() =>
+                        fetchAllMigrations().pipe(
+                            catchError((error): [ErrorLike] => [asError(error)]),
+                            repeatWhen(observable => observable.pipe(delay(REFRESH_INTERVAL_MS)))
+                        )
+                    ),
+                    takeWhile(() => true, true)
```
Comment (eseliger) [inline client/web/src/site-admin/SiteAdminMigrationsPage.tsx:84]: arent there two timers? One from `timer` and one from `delay`?

File: client/web/src/site-admin/SiteAdminMigrationsPage.scss [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        &__grid {
+            display: grid;
+            grid-template-columns: [date] max-content [error] minmax(auto, 1fr) [end];
+            align-items: center;
+
+            &-code {
+                border-left: 1px solid var(--border-color-2);
+            }
+        }
+    }
```
Comment (eseliger) [inline client/web/src/site-admin/SiteAdminMigrationsPage.scss:46]: this doesn't follow BEM, so eslint should fail.

I think you can do something like `.site-admin-migration-node-errors__grid`

File: client/web/src/site-admin/SiteAdminMigrationsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+
+/** Returns the searchable text from a migration. */
+const searchFields = (migration: OutOfBandMigrationFields): string[] => [
+    migration.team,
+    migration.component,
+    migration.description,
+    migration.introduced,
+    migration.deprecated,
+    ...migration.errors.map(error => error.message),
+]
```
Comment (eseliger) [inline client/web/src/site-admin/SiteAdminMigrationsPage.tsx:423]: This fails compilation, maybe this helps?

```suggestion
const searchFields = (migration: OutOfBandMigrationFields): string[] => [
    migration.team,
    migration.component,
    migration.description,
    migration.introduced,
    migration.deprecated,
    ...migration.errors.map(error => error.message),
].filter((value: string | null): value is string => !!value)
```

**Review by eseliger — APPROVED** [review]
Nice 🙏

File: client/web/src/site-admin/SiteAdminMigrationsPage.tsx (hunk 2 above)
Comment (efritz) [reply client/web/src/site-admin/SiteAdminMigrationsPage.tsx:85]: I'm going to go through and fix this in the other two code intel pages as well in another PR.

**Review by unknwon — APPROVED** [review]
Oh wow, looking nice!!

**Pull request merged**